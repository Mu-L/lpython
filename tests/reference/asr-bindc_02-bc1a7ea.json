{
    "basename": "asr-bindc_02-bc1a7ea",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/bindc_02.py",
    "infile_hash": "ce0bbd7a1a17c689c995fb31bc71275c59459120f0cf7defedd1cf33",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-bindc_02-bc1a7ea.stdout",
    "stdout_hash": "80091a70711348b6d323cd569534e2f245806ac4ed3988a9c29577a8",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}