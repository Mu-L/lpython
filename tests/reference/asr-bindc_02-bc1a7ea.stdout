(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            __main__global_stmts:
                                (Function
                                    (SymbolTable
                                        211
                                        {
                                            
                                        })
                                    __main__global_stmts
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [f]
                                    []
                                    [(CPtrToPointer
                                        (Var 2 queries)
                                        (Var 2 x)
                                        (ArrayConstant
                                            4
                                            [1]
                                            (Array
                                                (Integer 4)
                                                [((IntegerConstant 0 (Integer 4) Decimal)
                                                (IntegerConstant 1 (Integer 4) Decimal))]
                                                PointerToDataArray
                                            )
                                            RowMajor
                                        )
                                        (ArrayConstant
                                            4
                                            [0]
                                            (Array
                                                (Integer 4)
                                                [((IntegerConstant 0 (Integer 4) Decimal)
                                                (IntegerConstant 1 (Integer 4) Decimal))]
                                                PointerToDataArray
                                            )
                                            RowMajor
                                        )
                                    )
                                    (Print
                                        (StringFormat
                                            ()
                                            [(Var 2 queries)
                                            (Var 2 x)]
                                            FormatPythonFormat
                                            (String 1 () ExpressionLength CString)
                                            ()
                                        )
                                    )
                                    (SubroutineCall
                                        2 f
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            f:
                                (Function
                                    (SymbolTable
                                        210
                                        {
                                            y:
                                                (Variable
                                                    210
                                                    y
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Integer 2)
                                                        [((IntegerConstant 0 (Integer 4) Decimal)
                                                        (IntegerConstant 2 (Integer 4) Decimal))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            yptr1:
                                                (Variable
                                                    210
                                                    yptr1
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Pointer
                                                        (Array
                                                            (Integer 2)
                                                            [(()
                                                            ())]
                                                            DescriptorArray
                                                        )
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            yq:
                                                (Variable
                                                    210
                                                    yq
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (CPtr)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                )
                                        })
                                    f
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(Assignment
                                        (Var 210 yq)
                                        (PointerNullConstant
                                            (CPtr)
                                        )
                                        ()
                                        .false.
                                    )
                                    (Assignment
                                        (Var 210 y)
                                        (ArrayBroadcast
                                            (ArrayItem
                                                (Var 210 y)
                                                [(()
                                                (IntegerConstant 0 (Integer 4) Decimal)
                                                ())]
                                                (Integer 2)
                                                RowMajor
                                                ()
                                            )
                                            (ArrayConstant
                                                4
                                                [2]
                                                (Array
                                                    (Integer 4)
                                                    [((IntegerConstant 1 (Integer 4) Decimal)
                                                    (IntegerConstant 1 (Integer 4) Decimal))]
                                                    FixedSizeArray
                                                )
                                                ColMajor
                                            )
                                            (Array
                                                (Integer 2)
                                                [((IntegerConstant 0 (Integer 4) Decimal)
                                                (IntegerConstant 2 (Integer 4) Decimal))]
                                                FixedSizeArray
                                            )
                                            ()
                                        )
                                        ()
                                        .false.
                                    )
                                    (Assignment
                                        (ArrayItem
                                            (Var 210 y)
                                            [(()
                                            (IntegerConstant 0 (Integer 4) Decimal)
                                            ())]
                                            (Integer 2)
                                            RowMajor
                                            ()
                                        )
                                        (Cast
                                            (IntegerConstant 1 (Integer 4) Decimal)
                                            IntegerToInteger
                                            (Integer 2)
                                            (IntegerConstant 1 (Integer 2) Decimal)
                                        )
                                        ()
                                        .false.
                                    )
                                    (Assignment
                                        (ArrayItem
                                            (Var 210 y)
                                            [(()
                                            (IntegerConstant 1 (Integer 4) Decimal)
                                            ())]
                                            (Integer 2)
                                            RowMajor
                                            ()
                                        )
                                        (Cast
                                            (IntegerConstant 2 (Integer 4) Decimal)
                                            IntegerToInteger
                                            (Integer 2)
                                            (IntegerConstant 2 (Integer 2) Decimal)
                                        )
                                        ()
                                        .false.
                                    )
                                    (Assignment
                                        (Var 210 yptr1)
                                        (GetPointer
                                            (Var 210 y)
                                            (Pointer
                                                (Array
                                                    (Integer 2)
                                                    [(()
                                                    ())]
                                                    DescriptorArray
                                                )
                                            )
                                            ()
                                        )
                                        ()
                                        .false.
                                    )
                                    (Print
                                        (StringFormat
                                            ()
                                            [(GetPointer
                                                (Var 210 y)
                                                (Pointer
                                                    (Array
                                                        (Integer 2)
                                                        [(()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                )
                                                ()
                                            )
                                            (Var 210 yptr1)]
                                            FormatPythonFormat
                                            (String 1 () ExpressionLength CString)
                                            ()
                                        )
                                    )
                                    (Print
                                        (StringFormat
                                            ()
                                            [(ArrayItem
                                                (Var 210 yptr1)
                                                [(()
                                                (IntegerConstant 0 (Integer 4) Decimal)
                                                ())]
                                                (Integer 2)
                                                RowMajor
                                                ()
                                            )
                                            (ArrayItem
                                                (Var 210 yptr1)
                                                [(()
                                                (IntegerConstant 1 (Integer 4) Decimal)
                                                ())]
                                                (Integer 2)
                                                RowMajor
                                                ()
                                            )]
                                            FormatPythonFormat
                                            (String 1 () ExpressionLength CString)
                                            ()
                                        )
                                    )
                                    (Assert
                                        (IntegerCompare
                                            (ArrayItem
                                                (Var 210 yptr1)
                                                [(()
                                                (IntegerConstant 0 (Integer 4) Decimal)
                                                ())]
                                                (Integer 2)
                                                RowMajor
                                                ()
                                            )
                                            Eq
                                            (Cast
                                                (IntegerConstant 1 (Integer 4) Decimal)
                                                IntegerToInteger
                                                (Integer 2)
                                                (IntegerConstant 1 (Integer 2) Decimal)
                                            )
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (Assert
                                        (IntegerCompare
                                            (ArrayItem
                                                (Var 210 yptr1)
                                                [(()
                                                (IntegerConstant 1 (Integer 4) Decimal)
                                                ())]
                                                (Integer 2)
                                                RowMajor
                                                ()
                                            )
                                            Eq
                                            (Cast
                                                (IntegerConstant 2 (Integer 4) Decimal)
                                                IntegerToInteger
                                                (Integer 2)
                                                (IntegerConstant 2 (Integer 2) Decimal)
                                            )
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (CPtrToPointer
                                        (Var 210 yq)
                                        (Var 210 yptr1)
                                        (ArrayConstant
                                            4
                                            [2]
                                            (Array
                                                (Integer 4)
                                                [((IntegerConstant 0 (Integer 4) Decimal)
                                                (IntegerConstant 1 (Integer 4) Decimal))]
                                                PointerToDataArray
                                            )
                                            RowMajor
                                        )
                                        (ArrayConstant
                                            4
                                            [0]
                                            (Array
                                                (Integer 4)
                                                [((IntegerConstant 0 (Integer 4) Decimal)
                                                (IntegerConstant 1 (Integer 4) Decimal))]
                                                PointerToDataArray
                                            )
                                            RowMajor
                                        )
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            queries:
                                (Variable
                                    2
                                    queries
                                    []
                                    Local
                                    (PointerNullConstant
                                        (CPtr)
                                    )
                                    (PointerNullConstant
                                        (CPtr)
                                    )
                                    Default
                                    (CPtr)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                    .false.
                                    .false.
                                    ()
                                    .false.
                                    .false.
                                ),
                            x:
                                (Variable
                                    2
                                    x
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Pointer
                                        (Array
                                            (Integer 2)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                    .false.
                                    .false.
                                    ()
                                    .false.
                                    .false.
                                )
                        })
                    __main__
                    [numpy]
                    ()
                    .false.
                    .false.
                ),
            lpython_builtin:
                (IntrinsicModule lpython_builtin),
            main_program:
                (Program
                    (SymbolTable
                        212
                        {
                            __main__global_stmts:
                                (ExternalSymbol
                                    212
                                    __main__global_stmts
                                    2 __main__global_stmts
                                    __main__
                                    []
                                    __main__global_stmts
                                    Public
                                )
                        })
                    main_program
                    [__main__]
                    [(SubroutineCall
                        212 __main__global_stmts
                        2 __main__global_stmts
                        []
                        ()
                    )]
                ),
            numpy:
                (Module numpy)
        })
    []
)
